backend/src/main/java/eu/bbmri_eric/negotiator/user/PersonServiceImpl.java

```diff
@@ -135,6 +135,7 @@ public void assignAsRepresentativeForResource(Long representativeId, Long resour
       eventPublisher.publishEvent(
           new FirstRepresentativeEvent(this, resource.getId(), resource.getSourceId()));
     }
+    eventPublisher.publishEvent(new RepresentativeAddedEvent(this, resourceId, representativeId));
```

Comment:
I would put this after the changes are persisted, what if there is an error during persistence?